### PR TITLE
Feature: Re-enable login button if user enters wrong username and/or password [LEI-238]

### PR DESCRIPTION
**Pull request by samanehsan**
Refs: https://openscience.atlassian.net/browse/LEI-238
Companion to: https://github.com/CenterForOpenScience/exp-addons/pull/159
## Purpose

When a user enters a username and password and clicks the "Continue" button to login, the button is disabled during authentication. However, if either the username or password is invalid the button remains disabled and the user is unable to login after correcting their username/password.
## Changes
- If authentication returns a 404 error, set `this.authenticating` to false
- Pass `invalidAuth` to the component. Necessary for the [companion PR](https://github.com/CenterForOpenScience/exp-addons/pull/159), which will disable the continue button when the invalidAuth modal is open

### CODE REVIEW

File: app/components/isp-jam-auth/template.hbs [hunk 1]
```diff
@@ -13,7 +13,7 @@
         </div>
       </div>
       <div class="form-group">
-        <button disabled={{if (or authenticating (not (and password username))) true false}} class="btn btn-primary" onclick={{action 'authenticate'}}>{{t 'global.continueLabel'}}</button>
+        <button disabled={{disabled}} class="btn btn-primary" onclick={{action 'authenticate'}}>{{t 'global.continueLabel'}}</button>
```
Comment (abought) [inline app/components/isp-jam-auth/template.hbs]: We generally encourage slightly more descriptive variable names. (like `allowLogin` or `loginEnabled` or `NScanClickTheShinyButtonZOMGSoShiny`*)

For example, if we ever wanted to have other UI in that component that could also be toggled, the word "disabled" could quickly become ambiguous.

File: app/templates/participate/login.hbs [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <div class="row">
       <div class="col-md-12">
         <h1>International Situations Project</h1>
-        {{isp-jam-auth login=(action 'authenticate')}}
+        {{isp-jam-auth invalidAuth=invalidAuth authenticating=authenticating login=(action 'authenticate')}}
```
Comment (abought) [inline app/templates/participate/login.hbs:27]: I can't seem to find where the `isp-jam-auth` component uses the `invalidAuth` variable- can you please clarify why this needs to be passed in?

File: app/templates/participate/login.hbs [hunk 3]
```diff
@@ -24,7 +24,7 @@
     <div class="row">
       <div class="col-md-12">
         <h1>International Situations Project</h1>
```
Comment (abought) [inline app/templates/participate/login.hbs:26]: Although not part of this PR- should the title have a translation string? Or are we intentionally keeping that hard-coded in english?

File: app/components/isp-jam-auth/template.hbs (hunk 1 above)
Comment (abought) [reply app/components/isp-jam-auth/template.hbs]: Can you please clarify where this property is defined? It does not appear to be part of the matching component.js, nor is it a variable passed in.

File: app/controllers/participate/login.js [hunk 4]
```diff
@@ -25,6 +26,7 @@ export default Ember.Controller.extend({
         .catch((e) => {
           if (e.status === 404) {
             this.send('toggleInvalidAuth');
+            this.set('authenticating', false);
```
Comment (abought) [inline app/controllers/participate/login.js:29]: Oddly, I also can't find where `authenticating` ever gets set to `true`... am I going blind today?

**Review by abought — COMMENTED** [review]
A few review questions before moving forward.

File: app/templates/participate/login.hbs (hunk 3 above)
Comment (samanehsan) [reply app/templates/participate/login.hbs:26]: This is supposed to stay in English, which is why it's hard-coded

File: app/components/isp-jam-auth/template.hbs (hunk 1 above)
Comment (samanehsan) [reply app/components/isp-jam-auth/template.hbs]: `isp-jam-auth` extends `jam-auth`, where all of those properties are defined: https://github.com/CenterForOpenScience/exp-addons/blob/develop/exp-models/addon/components/jam-auth/component.js

File: app/controllers/participate/login.js (hunk 4 above)
Comment (samanehsan) [reply app/controllers/participate/login.js:29]: That happens in https://github.com/CenterForOpenScience/exp-addons/blob/develop/exp-models/addon/components/jam-auth/component.js#L17

File: app/templates/participate/login.hbs (hunk 2 above)
Comment (samanehsan) [reply app/templates/participate/login.hbs:27]: It's used in https://github.com/CenterForOpenScience/exp-addons/blob/develop/exp-models/addon/components/jam-auth/component.js#L17 and it's passed in so that the button remains disabled while the invalid auth modal is open (and then it will be re-enabled when the modal is dismissed)

**Comment by samanehsan** [comment]
@abought, I renamed the `disabled` property to `allowLogin` and added the disabled attribute to the parent component's template.
